pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -321,13 +326,153 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 		// Due to immutable fields `spec.volumes`, pod update events are ignored.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
 		// A new Node may make a pod schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}, QueueingHintFn: pl.isSchedulableAfterNodeAdded},
 		// Pods may fail to schedule because the PVC it uses has not yet been created.
 		// This PVC is required to exist to check its access modes.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterPersistentVolumeClaimChange},
+	}
+}
+
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted,
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	deletedPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPodDeleted: %w", err)
+	}
+
+	if deletedPod.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	newPodPvcs, err := pl.readWriteOncePodPVCsForPod(pod, false)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.V(5).Info("no PVC for the Pod is found, this Pod won't be schedulable until PVC is created", "pod", klog.KObj(pod), "err", err)
+			return framework.QueueSkip, nil
+		}
+		return framework.Queue, err
+	}
+
+	// deletedPod may contain multiple PVCs and one or more PVCs have been deleted (but some still exist),
+	// so we need to ignore not found error,
+	// because we also need to determine if the undeleted pvc relate to the current pod or not.
+	deletedPodPvcs, err := pl.readWriteOncePodPVCsForPod(deletedPod, true)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	// If oldPod and the current pod are in conflict because of readWriteOncePodPVC,
+	// the current pod may be scheduled in the next scheduling cycle, so we return Queue
+	for pvc := range deletedPodPvcs {
+		if newPodPvcs.Has(pvc) {
+			return framework.Queue, nil
+		}
+	}
+
+	// If the pod has a readWriteOncePodPVC conflict,
+	// we should return QueueSkip, since it is not schedulable until the readWriteOncePodPVC conflict is resolved
+	for pvc := range newPodPvcs {
+		key := framework.GetNamespacedName(pod.Namespace, pvc)
+		if pl.sharedLister.StorageInfos().IsPVCUsedByPods(key) {
+			return framework.QueueSkip, nil
+		}
+	}
```

Comment:
L375 - L382 checks if no PVCs in the Pod have readWriteOncePodPVC conflict, right? 
Then, on second thought, do we need L359 - L373, which deletedPod had readWriteOncePodPVC-conflicting PVCs?

Reply:
Yes we don't have to check deletedPod has readWriteOncePodPVC conflict

Reply:
I think we still need L359 - L373

If L359 - L373 are deleted, then when the pvc of `deletedPod` conflicts with the current pod, we return `framework.QueueSkip` (this is reasonable, because if the current pod encounters a pvc conflict, regardless of whether the pvc is related to deletedPod, we should Skip this event), but if the user deletes the pvc after deleting the conflict pod, the current pod will not be scheduled again because we have not registered the deletion event of the pvc

So we need to determine whether the pvc of `deletedPod` conflicts with the current pod. If it conflicts, we return `framework.Queue` because it is likely to be scheduled after the user deletes the pvc.

Reply:
Wait... I noticed we're making a big mistake here.
We cannot use `sharedLister` in QHint because there's no guarantee that `isSchedulableAfterPodDeleted` is called after `sharedLister` knows `deletedPod` is deleted.
Actually, sharedLister is updated every time the scheduling cycle is executed: 
https://github.com/sanposhiho/kubernetes/blob/master/pkg/scheduler/schedule_one.go#L391-L393

So, we probably have to chagne all QHints not to use `sharedLister` at all. We have no choice but to simplify QHint and to determine when to return Queue more roughly.

Reply:
Doc update to prevent this mistake... https://github.com/kubernetes/kubernetes/pull/122017

Reply:
After looking at some of the code, I realized that we can't easily get a pod that conflicts because of `readWriteOncePodPVC` without using sharedLister. so we should need to delete the judgment related to `readWriteOncePodPVC` and modify the release-note

Reply:
May you take a look at the latest changes?